components/SpotifyInterface/SpotifyMakeRequest.c

```diff
@@ -32,222 +42,277 @@ bool Spotify_FindCode(char *Response, uint16_t SizeRes)
-    strcpy(Command, Command_t);
-    memset(Buf, 0x00, SizeBuf);
-    sprintf(Buf, "%s /v1/me/player/%s HTTP/1.1\r\n"
-                 "Host: api.spotify.com\r\n"
-                 "Authorization: Bearer %s\r\n"
-                 "Content-Length: 0\r\n"
-                 "Connection: close\r\n\r\n",
-            Method, Command, AccessToken);
-    ESP_LOGI(TAG, "\n\nrequest for %s \n%s\nand size it =%d\n\n",Command, Buf, strlen(Buf));
-    char URL[SMALL_BUF] = "https://api.spotify.com";
-    char Server[SMALL_BUF] = "api.spotify.com";
-    HttpsHandler(Buf, SizeBuf, URL, sizeof(URL), Server, sizeof(Server));
-}
+    // Enable detailed logging for debugging
+    // esp_http_client_set_debug(client, true);
 
-/**
- * @brief This function sends a request to the Spotify API to skip to the next track in the player.
- * @return This function does not return a value.
- */
-void Spotify_SendRequestForNext(Token_t *Token)
-{
-    char Buf[LONG_BUF];
-    ESP_LOGE(TAG, "we are in Send Spotify_SendRequestForNext ");
-    Spotify_MakePlayerCommandAndSendIt("POST", "next", Buf, LONG_BUF, Token->AccessToken);
-}
+    // Perform HTTP request
+    esp_err_t err = esp_http_client_perform(client);
 
-/**
- * @brief This function sends a request to the Spotify API to skip to the previous track in the player.
- * @return This function does not return a value.
- */
-void Spotify_SendRequestForPrevious(Token_t *Token)
-{
-    char Buf[LONG_BUF];
-    Spotify_MakePlayerCommandAndSendIt("POST", "previous", Buf, LONG_BUF, Token->AccessToken);
-}
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "HTTP client perform failed: %s", esp_err_to_name(err));
+    } else {
+        ESP_LOGI(TAG, "HTTP client performed successfully");
+    }
+    
+    // Cleanup HTTP client
+    esp_http_client_cleanup(client);
 
-/**
- * @brief This function sends a request to the Spotify API to play the current track in the player.
- * @return This function does not return a value.
- */
-void Spotify_SendRequestForPlay(Token_t *Token)
-{
-    char Buf[LONG_BUF];
-    Spotify_MakePlayerCommandAndSendIt("PUT", "play", Buf, LONG_BUF, Token->AccessToken);
 }
 
 /**
- * @brief This function sends a request to the Spotify API to pause the current track in the player.
- * @return This function does not return a value.
- */
-void Spotify_SendRequestForPause(Token_t *Token)
+* @brief This function sends a request to the Spotify API to perform a play/pause/next/previous commands.
+* @param[in] Command_ The specific player command to be executed (e.g., "next", "previous", "play", "pause").
+* @param[in] AccessToken given by authorization step.
+* @return This function does not return a value.
+*/
+void Spotify_ControlPlayback(int Command, char *AccessToken)
 {
-    char Buf[LONG_BUF];
-    Spotify_MakePlayerCommandAndSendIt("PUT", "pause", Buf, LONG_BUF, Token->AccessToken);
-}
+    // fill path and method variables based on the Spotify API document
+    char path[30] = {};
```

Comment:
please use "Pascal" format for naming .

Reply:
for local variables we use camelCase, change applied

Reply:
for local variables, we use camelCase
change appleid

Reply:
you just use camelCase for some Spotify file !, but we use Pascal in other file, and please write code in an integrated manner with the rest of the components .

Reply:
due to our discussion, we will fix other modules in the same way:
- camleCase for local variables
- PascalCase for global variables